ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Python3_Source/tasks/main.yml

```diff
@@ -0,0 +1,95 @@
+---
+##############
+# Python3_Source #
+##############
+- name: Test if python3 is installed on path
+  shell: python3 --version >/dev/null
+  ignore_errors: yes
+  register: python3_installed
+  tags:
+    - python3_source
+
+- name: Ensure zlib1g and libffi-dev is installed on Ubuntu before compiling python3
+  apt:
+    name: "{{ packages }}"
+  vars:
+    packages:
+    - zlib1g
+    - zlib1g-dev
+    - libffi-dev
+  when:
+    - (ansible_distribution == "Ubuntu")
+    - (python3_installed.rc != 0 )
+  tags: python3_source
+
+- name: Ensure zlib-devel and libffi-devel-gcc5 is installed on SLES before compiling python3
+  zypper: 
+    name: "{{ packages }}"
+  vars:
+    packages:
+    - zlib-devel
+    - libffi-devel-gcc5
+  when:
+    - (ansible_distribution == "SLES")
+    - (python3_installed.rc != 0 )
+  tags: python3_source
+
+- name: Ensure zlib-devel and libffi-devel is installed on CentOS before compiling python3
+  yum: 
+    name: "{{ packages }}"
+  vars:
+    packages:
+    - zlib-devel
+    - libffi-devel
+  when:
+    - (ansible_distribution == "CentOS")
```

Comment:
This task could run on CentOS and RHEL platforms. This way you can dispose of the task on RHEL: Ensure zlib-devel and libffi-devel is installed on RedHat before compiling python3 L49-59

Reply:
RHEL and CENT have been squashed into one task